lib/vector-core/src/metrics/mod.rs

```diff
@@ -111,6 +111,16 @@ pub fn capture_metrics(controller: &Controller) -> impl Iterator<Item = Event> {
         .iter()
         .map(|kv| Metric::from_metric_kv(kv.key().key(), kv.value()).into())
         .collect::<Vec<Event>>();
+
+    // Add alias `events_processed_total` for `events_out_total`.
+    for i in 0..events.len() {
+        let metric = events[i].as_metric();
+        if metric.name() == "events_out_total" {
+            let alias = metric.clone().with_name("events_processed_total");
+            events.push(alias.into());
+        }
+    }
```

Comment:
It would be nice to do this without using indexing, but I don't see a way without allocating a temporary vec.

Reply:
Yep.